### DISCUSSION TITLE
Add shadcn/ui... until we're ready to build

### DISCUSSION THREAD

**Discussion by andrecasal** [epicweb-dev/epic-stack#226] [Ideas]
In line with https://github.com/epicweb-dev/epic-stack/discussions/170, I think it would make sense to add all the necessary components we need to build a full-blown UI 😊

As it is now, devs need to use the shadcn/ui's CLI to add new shadcn/ui components. But these newly installed components aren't ready to be used. We still need to style them using the Epic Stack's custom Tailwind classes. This is something that needs to be done for every component we add. So we might as well include them already and give devs all the building blocks they need to hit the ground running.

What I propose is that we add all of shadcn/ui's components, already styled with Epic Stack's custom classes and add a few other ones I have in mind that every app needs (container, background effect, and a few other ones not included in shadcn/ui).

This has the added benefit of not having to use/learn about shadcn/ui's CLI and we could also remove the components.json file.

**Comment by kentcdodds** [comment]
> We still need to style them using the Epic Stack's custom Tailwind classes

This is not correct. I removed the custom classes in favor of the shadcn ones. I don't want to have a ton of components people don't need. So we're going to stick with what we've got. It doesn't take long to install a new component and use it.

**Reply by andrecasal** [reply]
Oh, I didn't notice you went back on that. Alright then 👍
